lib/segment/src/problems/unindexed_field.rs

```diff
@@ -0,0 +1,315 @@
+use std::collections::{HashMap, HashSet};
+use std::sync::OnceLock;
+use std::time::Duration;
+
+use http::{HeaderMap, Method, Uri};
+use issues::{Action, CodeType, ImmediateSolution, Issue, Solution};
+use itertools::Itertools;
+
+use crate::common::operation_error::OperationError;
+use crate::data_types::text_index::{TextIndexParams, TextIndexType, TokenizerType};
+use crate::json_path::{JsonPathInterface, JsonPathV2};
+use crate::types::{
+    AnyVariants, Condition, FieldCondition, Filter, Match, MatchValue, PayloadFieldSchema,
+    PayloadKeyType, PayloadSchemaParams, PayloadSchemaType,
+};
+
+#[derive(Debug)]
+pub struct UnindexedField {
+    field_name: JsonPathV2,
+    field_schemas: HashSet<PayloadFieldSchema>,
+    collection_name: String,
+    endpoint: Uri,
+}
+
+impl UnindexedField {
+    pub fn slow_search_threshold() -> Duration {
+        static SLOW_SEARCH_THRESHOLD: OnceLock<Duration> = OnceLock::new();
+
+        *SLOW_SEARCH_THRESHOLD.get_or_init(|| {
+            Duration::from_millis(
+                std::env::var("QDRANT_SLOW_SEARCH_THRESHOLD")
+                    .ok()
+                    .and_then(|var| var.parse::<u64>().ok())
+                    .unwrap_or(300),
+            )
+        })
+    }
+
+    pub fn get_code(collection_name: &str, field_name: &JsonPathV2) -> CodeType {
+        format!("/{collection_name}/UNINDEXED_FIELD/{field_name}")
+    }
```

Comment:
Sorry, but I still don't like this string format because it  requires string based matching when automatically resolving issues. That's error prone.

Is there a good reason for not doing it in a strongly typed manner? Could that be a bad idea because various issues might be on wildly different levels, where a simple string could easily represent them all?

How about a flat structure where all issues are on the same level, having some constant type such as `UNINDEXED_FIELD` where each type can have their own respective metadata - a collection and field name in this case?

Reply:
I kinda share your thoughts on this... 

How about this: If we adhere to a REST-like representation, we can avoid headaches. In this case, a better code for this issue could be `/collections/{collection_name}/points/index/{field_name}/UNINDEXED_FIELD`

What makes me reluctant to have a flat structure is that it will want to have a different implementation of the `issues::Dashboard`, in which it becomes event-based, so that we don't pollute the codebase too much by separating the event from the handling. Maybe it is not a bad thing either, but it will require at least another PR.